clients/acm.d.ts

```diff
@@ -0,0 +1,350 @@
+import {Request} from '../lib/request';
+import {Response} from '../lib/response';
+import {AWSError} from '../lib/error';
+import {Service} from '../lib/service';
+declare class ACM extends Service {
+  /**
```

Comment:
This might not be a must-have,  but it would be nice to have code completion work when I'm binding params to a service client on construction. You can define a constructor here (perhaps something like `constructor(options?: ServiceConfigurationOptions & ServiceParams)`) and define an interface below like:
One way you could do this is have a constructor such as:
`constructor(options?: ServiceConfigurationOptions & ServiceParams)`
And then near the bottom of the file you can define:
```
interface ServiceParams {
    params?: Shapes.AddTagsToCertificateRequest &
                    Shapes.DeleteCertificateRequest &
                    ...
}
```
Since the `TSGenerator` has to iterate through every operation input shape anyway, it could just keep track of a list of all of the input shape names.